noxfiles/git_nox.py

```diff
@@ -0,0 +1,166 @@
+"""Contains the nox sessions utilities for git-related tasks"""
+
+import re
+from enum import Enum
+from typing import List, Optional
+
+from packaging.version import Version
+
+import nox
+
+RELEASE_BRANCH_REGEX = r"release-(([0-9]+\.)+[0-9]+)"
+RC_TAG_REGEX = r"{release_version}rc([0-9]+)"
+RELEASE_TAG_REGEX = r"(([0-9]+\.)+[0-9]+)"
+VERSION_TAG_REGEX = "{version}{tag_type}([0-9]+)"
```

Comment:
Curious why there is no need for `VERSION_TAG_REGEX` to be treated as raw but there is for the others (this has confused me in the past too sometimes so not necessarily anything to change or do)

Reply:
I need to spend a day re-learning all of that at some point, I end up just throwing pasta at the wall most of the time 😄